kyo-core/shared/src/main/scala/kyo/scheduler/IOPromise.scala

```diff
@@ -237,15 +273,18 @@ private[kyo] object IOPromise:
 
         def waiters: Int
         def interrupt[E2 >: E](v: Error[E2]): Pending[E, A]
+        def removeInterrupt(other: IOPromise[?, ?]): Pending[E, A]
```

Comment:
This new method is relatively expensive since it'll rebuild the pending list. It should be ok since, in the majority of cases, the list is quite small with typically two elements: one for interrupt and another for completion. The exception is the collection handling methods in `Fiber`, where `State` objects link to all forked fibers but it becomes available for GC right after the collection operation finishes. I think we don't need an early cleanup there.